interfaces/builtin/firewall_control_test.go

```diff
@@ -87,8 +88,15 @@ func (s *FirewallControlInterfaceSuite) TestUsedSecuritySystems(c *C) {
 	snippet, err = s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecuritySecComp)
 	c.Assert(err, IsNil)
 	c.Assert(snippet, Not(IsNil))
-	// connected plugs have a non-nil security snippet for kmod
-	snippet, err = s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecurityKMod)
```

Comment:
You don't want to copy the definer here or anywhere else. Just create the specification and use the generic specification methods to query the interface. Just call `spec.AddConnectedPlug` and look at what was done to the specification.